slacktee.sh

```diff
@@ -203,7 +203,8 @@ function send_message()
 					\"icon_url\": \"$icon_url\" $parseMode}"
 
 				post_result=$(curl -H "Authorization: Bearer $token" -H 'Content-type: application/json; charset=utf-8' -X POST -d "$json" https://slack.com/api/chat.postMessage 2> /dev/null)
-				if [ $? != 0 ]; then
+				post_ok="$(echo "$post_result" | awk 'match($0, /"ok":([^,}]+)/) {print substr($0, RSTART+5, RLENGTH-5)}')"
```

Comment:
Starting to wonder if we are tying our hands in complexity with using only shell scripts here. I understand the goal is to keep slacktee as cross-platform as possible, but consider:

1) python (well, at least python 2) is included in the majority of [OSes by default](https://unix.stackexchange.com/questions/24802/on-which-unix-distributions-is-python-installed-as-part-of-the-default-install) (although with a few exceptions re: BSD. Do we have users running slacktee w/o python installed? Do we know?)
2) We are using `bash`, not `sh`. Are we using any bash extensions that aren't cross-platform? I've no idea.

If we've got all this complex shell script, I just want to make sure we are truly gaining something (cross-platform) that is actually needed (do our users need this)? I'm afraid there's like, 2 or 3 people in the world that understand  this script :)

Reply:
We could potentially do a binary with go, that is built per distro. Any dependencies can be baked in (static binary).

Reply:
Ah, I see now.

I would hesitate to use golang for this, just because it seems making a CLI tool in Python would be simpler, and more people could easily contribute.

But I do like the idea of installing missing deps (ie: python) w/ Ansible if missing. However, I think that might make integrating the script harder? right now you just drop the shell file on any system you want - idk how Ansible works, would it be that simple?

Reply:
I might be too stubborn but I would like to avoid using other languages in slacktee. 
slacktee is intentionally written in the shell script for the simple installation and the better portability. I agree python is now available in majority of OSes by default, but some people are using `slacktee` in bare minimum environments (see #37 and #57) and I cannot imaging that they are happily installing python just for using `slacktee`. 

With regard to point 2, I'm not sure if we are using non-cross-platfrom bash extensions. In my understanding, we are currently using `[[ ]]` for using regex in the if condition. Some OS like `alpine` doesn't have `bash` by default and using `sh` might be better for the portability. 
If we drop the support of conditional formatting, we may be able to make `slacktee` a `sh` script ;)

Reply:
> If we drop the support of conditional formatting, we may be able to make slacktee a sh script ;)

The opposite intention of my comment :P

To be more extreme, I wonder if there is a way to transpile a restrictive subset of some higher level language into a POSIX compliant sh script. Basically, I think it'd be awesome to have an easier language to develop in, but keep the awesome portability you want to uphold. An idea for another day :)

Reply:
To be clear, I am not opposing to implement `slacktee` functionalities in other languages.
If we start another project which is implementing slacktee in different languages, I would love to contribute!